src/payment/coupon/Benefit.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-i18n';
+
+import messages from './messages';
+import { PERCENTAGE_BENEFIT, ABSOLUTE_BENEFIT } from './data/constants';
+
+const Benefit = (props) => {
+  const {
+    type, code, value, intl,
+  } = props;
+
+  let formattedValue = null;
+  let formattedMessage = null;
+  switch (type) {
+    case PERCENTAGE_BENEFIT:
+      formattedValue = intl.formatNumber(value / 100, {
+        style: 'percent',
+      });
+      formattedMessage = intl.formatMessage(messages['payment.coupon.benefit.percentage'], {
+        code,
+        value: formattedValue,
+      });
+      break;
+    case ABSOLUTE_BENEFIT:
+      formattedValue = intl.formatNumber(value, {
+        style: 'currency',
+        currency: 'USD',
+        minimumFractionDigits: 0,
+      });
+      formattedMessage = intl.formatMessage(messages['payment.coupon.benefit.absolute'], {
+        code,
+        value: formattedValue,
+      });
+      break;
+    default:
```

Comment:
This default case seemed valuable to me; the alternative is showing nothing, which seems particularly bad fallback behavior.